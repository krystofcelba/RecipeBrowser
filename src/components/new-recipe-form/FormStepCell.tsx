import React from 'react';

import FormInputCell from './FormInputCell';
import { NewStep } from '../../redux/reducers/ui';
import i18n from '../../assets/i18n';

interface Props {
  step: NewStep;
  onChange: (step: NewStep) => void;
}

const FormStepCell = ({ step, onChange }: Props) => (
  <FormInputCell
    placeholder={i18n.t('step')}
    value={step.step}
    onChangeText={text => onChange({ ...step, step: text })}
    numberOfLines={3}
  />
);

export default FormStepCell;
