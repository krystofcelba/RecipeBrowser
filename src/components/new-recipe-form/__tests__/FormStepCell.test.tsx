import React from 'react';
import { shallow } from 'enzyme';

import FormStepCell from '../FormStepCell';

describe('FormStepCell component', () => {
  it('renders as expected', () => {
    const wrapper = shallow(<FormStepCell step={{ id: 1, step: 'test' }} onChange={() => {}} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('call onChange as expected', () => {
    const onChangeSpy = jest.fn();
    const wrapper = shallow(<FormStepCell step={{ id: 1, step: 'test' }} onChange={onChangeSpy} />);
    wrapper.simulate('changeText', 'a');
    expect(onChangeSpy.mock.calls).toMatchSnapshot();
  });
});
