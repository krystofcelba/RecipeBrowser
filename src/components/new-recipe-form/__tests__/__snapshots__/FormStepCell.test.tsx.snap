// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormStepCell component call onChange as expected 1`] = `
Array [
  Array [
    Object {
      "id": 1,
      "step": "a",
    },
  ],
]
`;

exports[`FormStepCell component renders as expected 1`] = `
<FormInputCell
  numberOfLines={3}
  onChangeText={[Function]}
  placeholder="Step"
  value="test"
/>
`;
